source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc

```diff
@@ -119,9 +119,16 @@ JsonTranscoderConfig::JsonTranscoderConfig(
     }
     for (int i = 0; i < service->method_count(); ++i) {
       auto method = service->method(i);
-      if (!PathMatcherUtility::RegisterByHttpRule(pmb,
-                                                  method->options().GetExtension(google::api::http),
-                                                  ignored_query_parameters, method)) {
+      auto http_rule = method->options().GetExtension(google::api::http);
+
+      if (!http_rule.pattern_case() && proto_config.auto_mapping()) {
+        auto post = "/" + service->file()->package() + "." + service->name() + "/" + method->name();
```

Comment:
`service->full_name()` should be good for package + "." + service->name.

Reply:
it seems the `service->full_name()` returns `bookstore.Bookstore.DeleteShelf` and we need `bookstore.Bookstore/DeleteShelf`.